src/main/resources/public/js/log.provider.js

```diff
@@ -2,11 +2,10 @@
   'use strict';
   angular.module('app').config(logProvider);
 
-  logProvider.$inject = [ '$logProvider', '$provide' ];
-
-  function logProvider($logProvider, $provide) {
-    $logProvider.debugEnabled(true);
+  logProvider.$inject = [ '$logProvider', '$provide', 'xmlRequestProvider' ];
 
+  function logProvider($logProvider, $provide, xmlRequestProvider) {
+    $logProvider.debugEnabled(xmlRequestProvider.$get().synchronousRequest().debug);
```

Comment:
First, the idea was to enable or disable the debugmode by url lik ?debug=true. But this is not possible.

A) We use ng-strict-di which means that any injected dependency will be checked if one instance is currently not created. In any cases the app will fail.

B) In `config` the $logProvider is instantiated only. But the config is called just once, so you need to use the `run` section but services only can be injected...

C) The solution is explained here: https://vasya10.wordpress.com/2014/10/13/angularjs-startup-time-environment-configuration/